Quaver.Shared/Screens/Gameplay/UI/PauseScreen.cs

```diff
@@ -135,9 +139,34 @@ public override void Update(GameTime gameTime)
             if (Screen.Failed)
                 Visible = false;
 
+            HandleInput();
+
             base.Update(gameTime);
         }
 
+        /// <summary>
+        ///     Handles all input for the screen.
+        /// </summary>
+        private void HandleInput()
+        {
+            HandleKeyPressEsc();
+        }
+        
+        /// <summary>
+        ///     Handles when the user presses ESC.
+        /// </summary>
+        private void HandleKeyPressEsc()
+        {
+            if (!Screen.IsPaused)
+                return;
+
+            if (KeyboardManager.IsUniqueKeyPress(Keys.Escape))
+                return;
+            
+            if (ChatManager.IsActive)
```

Comment:
Use `DialogManager.Dialogs.Count != 0`. This will handle the case of all dialog screens - not just the chat.

When moving it over to GameplayScreen, you can have this check at the top of the HandleInput method.

Reply:
When dismissing via the DialogManager, should I dismiss only the current screen, or all?